docs/getting-started.md

```diff
@@ -0,0 +1,260 @@
+for (uint160 sender = 0; sender < type(uint160).max; sender++) {
+    for (uint160 receiver = 0; receiver < type(uint160).max; receiver++) {
+        for (uint256 amount = 0; amount < type(uint256).max; amount++) {
+            check_transfer(address(sender), address(receiver), amount);
+        }
+    }
+}
+```
+
+[^symbolic-execution]: Note that the number of possible input combinations in our example is `2^160 * 2^160 * 2^256`, and it is computationally infeasible to actually run all of them individually. As a solution, symbolic testing employs the symbolic execution technique, which enables testing all the input combinations without actually running them individually.
+
+**Tips:**
+
+- Instead of declaring symbolic input parameters, you can dynamically create symbols inside the test using the Halmos cheatcodes. For instance, our running example can be rewritten as follows:
+  ```solidity
+  function check_transfer() {
+      address sender = svm.createAddress("sender");
+      address receiver = svm.createAddress("receiver");
+      uint256 amount = svm.createUint256("amount");
+      ...
+  }
+  ```
+
+- Halmos requires dynamically-sized arrays (including `bytes` and `string`) to be given with a fixed size. Thus they cannot be declared as input parameters, but need to be programmatically constructed. For example, a byte array can be generated using the `svm.createBytes()` cheatcode as follows:
+  ```solidity
+  bytes memory data = svm.createBytes(96, 'data');
+  ```
+  Similarly, a dynamic array of integers can be created as shown below:
+  ```solidity
+  uint256[] memory arr = new uint256[3];
+  for (uint i = 0; i < 3; i++) {
+      arr[i] = svm.createUint256('element');
+  }
+  ```
+  We are planning to implement more cheatcodes and features that can make it easier to declare or create dynamic arrays.
+
+### 2.2 Specify input conditions
+
+Recall that symbolic tests take into account all possible input combinations. However, not all input combinations are relevant or valid for every test scenario. Similar to fuzz tests, you can use `vm.assume()` to specify the conditions for valid inputs.
+
+In our example, the conditions for the valid sender and receiver addresses are specified as follows:
+```solidity
+vm.assume(receiver != address(0));
+vm.assume(token.balance(sender) >= amount);
+```
+Like fuzz tests, any input combinations that don't satisfy the `assume()` conditions are disregarded. This means that, after executing the above `assume()` statements, only the input combinations in which the receiver is non-zero and the sender has sufficient balance are considered. Other input combinations that violate these conditions are ignored.
+
+**Tips:**
+
+- You need to be careful not to exclude valid inputs by setting too strong input conditions.
+
+- In symbolic tests, avoid using `bound()` as it tends to perform poorly. Instead, use `vm.assume()` which is more efficient and easier to read and write.
```

Comment:
I would give an example like this:

```
uint256 tokenId = svm.createSymbolicUint256("tokenId");

// ❌ don't do this
tokenId = bound(tokenId, 1, MAX_TOKEN_ID);

// ✅ do this
vm.assume(1 <= tokenId && tokenId <= MAX_TOKEN_ID);
```

unfortunaly the bad pattern is more concise than our recommended pattern, maybe we want to introduce `svm.bound(val, min, max)`?